internal/runner/service.go

```diff
@@ -182,13 +183,22 @@ func (r *runnerService) Execute(srv runnerv1.RunnerService_ExecuteServer) error
 		return err
 	}
 
+	// This goroutine will be closed when the handler exits or earlier.
 	go func() {
-		defer func() {
-			_ = stdinWriter.Close()
-		}()
+		defer func() { _ = stdinWriter.Close() }()
 
 		if len(req.InputData) > 0 {
-			if _, err := stdinWriter.Write(req.InputData); err != nil {
+			_, err := stdinWriter.Write(req.InputData)
+
+			// When TTY is false, the service assumes that there will be no more input data and
```

Comment:
@mxsdev please verify if this assumption is correct. I need to close the `stdin` writer at some point and as we agreed that the client won't close the send direction, I need some other indication.

Reply:
Ugh, last second fixes. I changed that in https://github.com/stateful/runme/pull/134/commits/045466cad9f443d221f20f9635bfbe2e381cbfec.

> Other than that, yeah this is fine, but it will disable to ability to send stdin at all for non-PTY for any clients. Hopefully that's alright.

I can imagine a scenario when one wants to execute some command and provide some input through STDIN without starting a TTY. I tested it that way before. I will figure out something tomorrow.

Reply:
Gotcha. Maybe in a follow up pr?